src/coreclr/vm/vars.hpp

```diff
@@ -298,15 +298,15 @@ class REF : public OBJECTREF
 
 };
 
-// the while (0) syntax below is to force a trailing semicolon on users of the macro
-#define VALIDATEOBJECTREF(objref) do {if ((objref) != NULL) (objref).Validate();} while (0)
-#define VALIDATEOBJECT(obj) do {if ((obj) != NULL) (obj)->Validate();} while (0)
-
 #define ObjectToOBJECTREF(obj)     (OBJECTREF(obj))
 #define OBJECTREFToObject(objref)  ((objref).operator-> ())
 #define ObjectToSTRINGREF(obj)     (STRINGREF(obj))
 #define STRINGREFToObject(objref)  (*( (StringObject**) &(objref) ))
 
+// the while (0) syntax below is to force a trailing semicolon on users of the macro
+#define VALIDATEOBJECT(obj) do {if ((obj) != NULL) (obj)->Validate();} while (0)
+#define VALIDATEOBJECTREF(objref) do { VALIDATEOBJECT(OBJECTREFToObject(objref)); } while (0)
```

Comment:
How is this fixing the race condition? I would expect to see a temporary local variable or something like that as part of the fix.

Reply:
Sigh, you are right, I have somehow forgotten that this is a macro and not a function call and so the OBJECTREFToObject(objref) will get propagated into the other macro as is.